### ISSUE TITLE
Update cell value during inline edit.

### ISSUE THREAD

**Issue by EEMaddy** [akveo/ng2-smart-table#649]
Hi,

There are two fields (Code, Name) for which I'm performing an add operation (default mode set). When I select "Code" field using the completor, the corresponding "Name" Field should be populated automatically and they are 1:1 mapped. How can I do this?? I tried using custom editors but i failed.

Spent almost 4 days :( . Can anyone please show me a pointer. Thanks in advance.

----------------------
Code,        Name
----------------------
123,          China
989,          USA

**Comment by thovo** [comment]
I think you can do it with a custom editor in one column, make a custom editor component for this case, with a select box for code and a field to display the name, the data should be something like this `{code: 123, name: 'China'}`. Whenever you select a code, you show the corresponding name.

Or do you really need to have 2 distinct columns, it made the problem become more complicated, I am trying to solve the same thing. If this is the case, I will try to return to you later.

**Comment by EEMaddy** [comment]
Dear @thovo , really appreciate your quick response. Unfortunately my requirement needs me to have the fields in two distinct columns.

One of the approach I tried was,

1. Wrote custom edit component for "CODE" field and used ng2-completor.
2. monitored the selected value in the main component.
3. When the selected value changed, I updated the default value of "NAME" field.
4. Refreshed Settings.

I could debug and verify that everything went fine, except couldn't see the change in the view.

Thanks for making your time. Have a nice day.!!

**Comment by thovo** [comment]
In this case, it becomes the problem of speaking between two components, I think so.

Did you try to save the data into `localStorage` and retrieve it when you pass into the name field with `ngOnInit`? 

I didn't try it yet because I don't figure out the way to handle too much data in the `localStorage`. Your case would be the same like mine here #650

**Comment by EEMaddy** [comment]
Nope, I used data services to store and retrieve data. Monitored the value change using ngOnChanges.
